server/data/data-hbase/src/main/java/org/apache/james/rrt/hbase/HBaseRecipientRewriteTable.java

```diff
@@ -149,17 +152,20 @@ private Mappings feedUserDomainMappingsList(HTableInterface table, String user,
 
     @Override
     protected Mappings mapAddress(String user, Domain domain) throws RecipientRewriteTableException {
+        return getApplicableMappingRow(user, domain)
+            .map(MappingsImpl::fromRawString)
+            .orElse(MappingsImpl.empty());
+    }
+
+    private Optional<String> getApplicableMappingRow(String user, Domain domain) throws RecipientRewriteTableException {
         HTableInterface table = null;
-        String mappings;
         try {
             table = TablePool.getInstance().getRecipientRewriteTable();
-            mappings = getMapping(table, user, domain);
-            if (mappings == null) {
-                mappings = getMapping(table, WILDCARD, domain);
-            }
-            if (mappings == null) {
-                mappings = getMapping(table, user, Domains.WILDCARD);
-            }
+            HTableInterface tableCopy = table;
```

Comment:
No, as the lambda complained about re-allocation, and I got an error... I was forced to do this...

Reply:
"forced"?

Reply:
I mean, the table variable might throw the exception handled in the catch block.
At the same time it needs to be visible in the final block.
And it could not be re-allocated due to the lambdas I added.
So I needed an extra variable.

(Or we can very well decide we don't want this refactoring!)

Reply:
ok, thx for the explanation

Reply:
You're welcome. I was actually very surprised myself...